### DISCUSSION TITLE
Inconsistent / confusing documentation of resolver incompatibility with anonymization

### DISCUSSION THREAD

**Discussion by tmo1** [DNSCrypt/dnscrypt-proxy#2135] [Potential issues]
I'm trying to figure out which resolvers that do adult content blocking ("ACB") are compatible with anonymization, and I'm rather confused.

1. [This list](https://dnscrypt.info/public-servers/) labels many ACB resolvers with "Warning: This server is incompatible with anonymization," including the CleanBrowsing and Adguard DNS DNSCrypt servers, and some, but not all, of the DNS for Family ones.
2. [This list](https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v3/parental-control.md), on the other hand, labels the Adguard and DNS for Family ones as incompatible, but not the CleanBrowsing ones.
3. [This page](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Anonymized-DNS#caveats) states that "Relaying is currently incompatible with cisco, as this resolver doesn't correctly implement DNSCrypt yet, and doesn't support large queries," but makes no mention of any other DNSCrypt resolvers incompatible with relaying. Are there any resolvers that are compatible with relaying but still incompatible with anonymization?

I can't find any discussion, beyond the brief note in the third list item above, of why a given resolver would or would not be compatible with anonymization. Any information and explanation is welcome.

**Comment by jedisct1** [comment]
Anonymization requires version 2 of the protocol, which is what all servers support. The only exception is Cisco, that only implements version 1. Workarounds to still hide client IP addresses with their servers used to exist, but intentionally or not, they eventually blocked them. It sounds safe to document this and unconditionally skip these servers when anonymization is requested. Things are different with other servers.

The other servers currently incompatible are:
- `adguard`
- `ams-dnscrypt-nl`
- `cs-lv`
- `cleanbrowsing`
- `dnscrypt.pl`
- `safesurfer`
- `yandex`

This is unintentional and either due to them running an old version of `dnsdist` that had a bug that has been fixed since, or to network configuration issues (blocking IP fragments, and that can happen at the cloud provider layer).

These incompatibilities are hopefully temporary. For example virtually all `cs-*` servers used to be incompatible but they eventually fixed the networking configuration and became compatible.

The `parental-control` and `opennic` lists are subsets of the `public-resolvers` list, and some descriptions may not be consistent. Feel free to report them or submit pull requests to the dedicated respository.

**Reply by tmo1** [reply]
Thank you for the explanation.

> Feel free to report them or submit pull requests to the dedicated respository.

[Done](https://github.com/DNSCrypt/dnscrypt-resolvers/issues/697).
